android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/RpcClient.java

```diff
@@ -278,7 +330,11 @@ public synchronized boolean authorize(String password) {
      * @return true if connected to BOINC core client, false if not connected
      */
     public final boolean isConnected() {
-        return (mSocket != null && mSocket.isConnected());
+        if (mIsRemote) {
```

Comment:
I'd check here for bot local are remote connection. What if GUI is already connected to client locally and try to establish new remote connection?

Reply:
You can run the boinc only once. and you can run it two possibility. once you run it lets say with remote, it connect with connect remote different with connect unix and save remote member. The program knows what state of the boinc is run, and no need to guess.

Reply:
ok. I push the change.